tests/specification_tests.rs

```diff
@@ -209,13 +210,18 @@ fn run_test_suite(tests_file: PathBuf) -> Result<(), Box<dyn Error>> {
 #[test]
 fn run_specification_tests() -> Result<(), Box<dyn Error>> {
     let test_suites_dir: PathBuf = env::current_dir()?.join("tests").join("test_suites");
-    for file in fs::read_dir(test_suites_dir)? {
-        let file_path = file?.path();
-        println!(
-            "\n## Test suite file: {:?}\n",
-            &file_path.file_name().unwrap()
-        );
-        run_test_suite(file_path)?
+    let json_files = fs::read_dir(test_suites_dir)?
+        .filter_map(Result::ok)
+        .filter(|fp| {
+            if let Some(ext) = fp.path().extension() {
```

Comment:
This can be simplified to `|fp| fp.path().extension() == Some("json")`

Reply:
It didn't :-(. So I had to either match explicitly, or unwrap 

 <img width="469" alt="Screenshot 2020-06-08 at 14 44 28" src="https://user-images.githubusercontent.com/10127682/84026827-acf2c000-a996-11ea-9d6a-6cde93a99f21.png">

Reply:
I'll use `|fp| fp.path().extension().unwrap_or_default() == "json"` 🙂